src/parser/jobs/sam/Buffs.js

```diff
@@ -0,0 +1,132 @@
+import React, {Fragment} from 'react'
+
+import {ActionLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+import Module from 'parser/core/Module'
+import {Rule, Requirement} from '../../core/modules/Checklist'
+import {Suggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+//import {TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+
+const JINPU_DURATION = 30000
+const SHIFU_DURATION = 30000
+
+const BUFF_BUFFER = 10000
+
+export default class Buffs extends Module {
+	static handle = 'buffs'
+	static dependencies = [
+		'checklist',
+		'combatants',
+		'invuln',
+		'suggestions',
+	]
+
+	_JinpuUses = []
+	_ShifuUses = []
+
+	_earlyJinpuApplications = 0
+	_earlyShifuApplications = 0
+
+	constructor(...args) {
+		super(...args)
+
+		const filter = {
+			by: 'player',
+			abilityId: [STATUSES.JINPU.id, STATUSES.SHIFU.id],
+		}
+		this.addHook(['applybuff', 'refreshbuff'], filter, event => {
+			this._onShifuApplication(event)
```

Comment:
This is still listening for jinpu _or_ shifu, and then running clip checks for _both_ - probably artificially inflating clip amount. Using existing code, this'll need to be more along the lines of

```js
this.addHook(['applybuff', 'refreshbuff'], {by: 'player', abilityId: STATUSES.JINPU.id}, this._onJinpuApplication)
this.addHook(['applybuff', 'refreshbuff'], {by: 'player', abilityId: STATUSES.SHIFU.id}, this._onShifuApplication)
```